### PR TITLE
Automatic notification test

### PR DESCRIPTION
**Pull request by scottinet**

### CODE REVIEW

File: test/api/core/notifier/checkNewRoutes.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    this.timeout(100);
+
+    Notifier.__with__({
+      notifyDocumentCreate: mockupAction,
+      notifyDocumentUpdate: mockupAction,
+      notifyDocumentDelete: mockupAction
+    })(function () {
+      for (var action in kuzzle.services.list.readEngine) {
+        if (kuzzle.services.list.readEngine.hasOwnProperty(action) && typeof kuzzle.services.list.readEngine[action] === 'function') {
+          knownActions.push(action);
```
Comment (AnthonySendra) [inline test/api/core/notifier/checkNewRoutes.test.js]: I'm not sure to understand why we also push functions without testing if the blacklist contains this function ?

**Comment by benoitvidis** [comment]
+1

**Comment by ballinette** [comment]
+1

File: test/api/core/notifier/checkNewRoutes.test.js (hunk 1 above)
Comment (scottinet) [inline test/api/core/notifier/checkNewRoutes.test.js]: It's to make the (knownActions - performedActions) array match exactly the blacklist one (line 66).

It allows testing the following cases with one should() test:

* if an ES function is not blacklisted (should fail)
* if an ES function throws a notification AND is blacklisted in the test (should fail)
* if an ES function has a notification and is not blacklisted (should succeed)

Comment (j33f) [inline test/api/core/notifier/checkNewRoutes.test.js]: +1


*Jeff - Kuzzle Team*
jeff@kuzzle.io

------------------------------
kuzzle.io
@kuzzleio <https://twitter.com/kuzzleio>
Fork it on GitHub <https://github.com/kuzzleio>

2015-11-04 17:57 GMT+01:00 Sébastien Cottinet <notifications@github.com>:

**Pull request merged**